org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ResolveClasspathHandler.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import static org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin.logException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.search.IJavaSearchConstants;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.SearchMatch;
+import org.eclipse.jdt.core.search.SearchParticipant;
+import org.eclipse.jdt.core.search.SearchPattern;
+import org.eclipse.jdt.core.search.SearchRequestor;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.ls.core.internal.ClasspathParams;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.lsp4j.jsonrpc.CompletableFutures;
+
+/**
+ * @author xuzho
+ *
+ */
+public class ResolveClasspathHandler {
+
+	public CompletableFuture<String> resolveClasspath(ClasspathParams param) {
+		return CompletableFutures.computeAsync(cm -> {
+			try {
+				return computeClassPath(param.getProjectName(), param.getStartupClass());
+			} catch (CoreException e) {
+				logException("Failed to resolve classpath.", e);
+				return null;
```

Comment:
took a look at the implementation of  [exceptionhandler of lsp4j](https://github.com/eclipse/lsp4j/blob/ffd31df92dcc91690d34c5f563754cdd2cb6c70a/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java), seems that should wrap the original exception into ResponseErrorException, otherwise the exception would be handled as server error